font/src/lib.rs

```diff
@@ -220,20 +220,25 @@ pub struct RasterizedGlyph {
     pub height: i32,
     pub top: i32,
     pub left: i32,
-    pub buf: Vec<u8>,
+    pub buf: BitmapBuffer,
 }
 
-impl Default for RasterizedGlyph {
-    fn default() -> RasterizedGlyph {
-        RasterizedGlyph { c: ' ', width: 0, height: 0, top: 0, left: 0, buf: Vec::new() }
-    }
+#[derive(Clone, Debug)]
+pub enum BitmapBuffer {
+    RGB(Vec<u8>),
+    RGBA(Vec<u8>),
 }
```

Comment:
I'm not sure if I've asked you this on IRC, but have you considered using `Vec<PixelFormat>` instead of `PixelFormat<Vec<u8>>`? That way each individual pixel will have the information attached to it.

Reply:
It wasn't discussed on IRC, but I've chosen this format for performance reasons and simplicity, since you need to pass these things to properly pass them to OpenGL renderer as a image pixels. So using Vec<PixelFormat> isn't playing well here.

Reply:
Oh right, the buffer is passed directly to OpenGL, specifying just the format using `gl::RGB` or `gl::RGBA`. I saw that at the end of my review, makes a lot of sense then.